src/DOM.js

```diff
@@ -1739,7 +1739,7 @@ class HTMLScriptElement extends HTMLLoadableElement {
       }
     });
     this.on('attached', () => {
-      if (this.getAttribute('src') && this.isRunnable() && this.isConnected && !this.readyState) {
+      if (this.isRunnable() && this.isConnected && !this.readyState) {
```

Comment:
It seems for the `innerHTML` run case, that should be handled by `runNow()` instead of `_loadRun` -- there is no (down)"load".

Rather than making `loadRunNow` handle a non-loading case, can we instead check here (on `attached`) which method we need to call (the loading one or not)?

Reply:
I think this change is the root of the fix. It ensures that the script will always be executed.

If anything, it might be a good idea to get rid of `runNow` and always go through a single path. That single path can check whether there's any URL.

Reply:
I'm fine with the single path, but the current state of this PR confuses things by re-adding innerHTML load under a second code path.

But folding `runNow` into a single code path that later checks the presence of the url would fix the underlying concern too.